Travel Scheduler/Test/Schedule.m

```diff
@@ -86,25 +80,47 @@ - (NSDictionary *)generateSchedule
         //NSLog([NSString stringWithFormat:@"Current Place: %@", currPlace.name]);  //TESTING only
         BOOL lockedPlaceAtCurrentTimeBlock = [self findAndSetLockedPlaceForDate:self.currDate time:self.currTimeBlock];
         if (lockedPlaceAtCurrentTimeBlock) {
-            NSNumber *cached = [currPlace.cachedTimeDistances objectForKey:self.currClosestPlace.placeId];
+            Commute *cached = [currPlace.cachedCommutes objectForKey:self.currClosestPlace.placeId];
             if (cached) {
-                self.currClosestPlace.travelTimeToPlace = cached;
+                self.currClosestPlace.commuteTo = cached;
+                self.currClosestPlace.travelTimeToPlace = cached.durationInSeconds;
             } else {
                 [self setTravelTimeFromOrigin:currPlace toPlace:self.currClosestPlace];
             }
         } else {
             [self getClosestAvailablePlace:currPlace atTime:self.currTimeBlock onDate:self.currDate];
         }
-        [self addAndUpdatePlace:dayPath atTime:self.currTimeBlock];
+        self.currClosestPlace.scheduledTimeBlock = self.currTimeBlock;
         if (self.currClosestPlace) {
             if (self.currClosestPlace.scheduledTimeBlock == getNextTimeBlock(currPlace.scheduledTimeBlock)) {
                 self.currClosestPlace.prevPlace = currPlace;
+            } else {
+                self.currClosestPlace.indirectPrev = currPlace;
+            }
+            BOOL placeFitsInSchedule = [self.currClosestPlace setArrivalDeparture:self.currTimeBlock];
```

Comment:
What is the difference between bool and BOOL?
If there is not difference, should we chose to adopt all the same one (because I usually use bool, but I can change it to BOOL)